integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/tcp/transport/test/util/NativeTCPClient.java

```diff
@@ -0,0 +1,184 @@
+package org.wso2.carbon.esb.tcp.transport.test.util;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.net.Socket;
+
+/**
+ * Java client for TCP transport.
+ */
+public class NativeTCPClient {
+
+    private String host = "localhost";
+    private int port = 6789;
+    private Socket socket = null;
+    private int messageCount = 0;
+    private String delimiterType;
+    private String stringDelimiter;
+    private Character byteDelimiter;
+    private String characterDelimiter;
+    private String message;
+
+    public NativeTCPClient(String delimiterType, int messageCount) throws Exception {
+        this.socket = new Socket(this.host, this.port);
+        this.messageCount = messageCount;
+        this.delimiterType = delimiterType;
+    }
+
+    public void sendToServer() throws Exception {
+        PrintWriter outToServer = null;
+        outToServer = new PrintWriter(new OutputStreamWriter(this.socket.getOutputStream()));
+        String fullMessage = "";
+        for (int i = 0; i < this.messageCount; ++i) {
+            fullMessage += this.message;
```

Comment:
Can use StringBuilder instead of concatenating

Reply:
Done